src/hotspot/share/gc/shared/gcLogPrecious.cpp

```diff
@@ -23,20 +23,16 @@
 
 #include "precompiled.hpp"
 #include "gc/shared/gcLogPrecious.hpp"
-#include "runtime/mutex.hpp"
-#include "runtime/mutexLocker.hpp"
+#include "runtime/semaphore.hpp"
 
 stringStream* GCLogPrecious::_lines = NULL;
 stringStream* GCLogPrecious::_temp = NULL;
-Mutex* GCLogPrecious::_lock = NULL;
+Semaphore* GCLogPrecious::_lock = NULL;
```

Comment:
Maybe renaming `_lock` to `_semaphore`? Additionally, since it's a binary semaphore, `new Semaphore(1)`, some comments explaining why `Mutex` is **not** suitable could avoid some future confusions.

PS: not a review, just a comment in passing.

Reply:
It's used as a lock, so I think the name `_lock` is appropriate. Instead I introduced a new class: `SemaphoreLock`, to make the code more readable (IMHO). Also added a comment. Hopefully, this addressed your comments.

Reply:
Thank you; it does look more readable. BTW, mutex has `try_lock` as well.

Reply:
I see. Then I'll remove that part of the new comment.